Make SQLiteTransaction notify SQLiteDatabaseTracker on macOS

bnham opened the pull request:
#### 722cf9996ceb152c7dfb9ba23235d0e456d29295
<pre>
Make SQLiteTransaction notify SQLiteDatabaseTracker on macOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=259062">https://bugs.webkit.org/show_bug.cgi?id=259062</a>
rdar://112027513

Reviewed by Chris Dumez and Brent Fulgham.

In 265791@main we enabled SQLiteDatabaseTracker on macOS, but it still does nothing because
SQLiteTransaction doesn&apos;t notify it. Change SQLiteTransaction to notify the tracker about ongoing
transactions on macOS as well.

* Source/WebCore/platform/sql/SQLiteTransaction.cpp:
(WebCore::SQLiteTransaction::begin):
(WebCore::SQLiteTransaction::commit):
(WebCore::SQLiteTransaction::rollback):
(WebCore::SQLiteTransaction::stop):

Canonical link: <a href="https://commits.webkit.org/265918@main">https://commits.webkit.org/265918@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63729d530d691bec4bdd458f512c05652355e1b6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12240 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12585 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12888 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13985 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11811 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/12270 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15003 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/12601 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14486 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/12404 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13197 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/14403 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10482 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11107 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18219 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11563 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11269 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14459 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11765 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-WK2-Tests-EWS "Waiting to run tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10981 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3015 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15311 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11621 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/sql/SQLiteTransaction.cpp:
```diff
@@ -58,15 +58,15 @@ void SQLiteTransaction::begin()
         // http://www.sqlite.org/lang_transaction.html
         // http://www.sqlite.org/lockingv3.html#locking
-#if PLATFORM(IOS_FAMILY)
+#if USE(RUNNINGBOARD)
```
Why do we even need those USE(RUNNINGBOARD) checks?

cdumez replied:
I don't think this code hurts anything on other platforms so I'd suggest not polluting the code with conditional checks.

bnham replied:
Guess I could remove all the `USE(RUNNINGBOARD)`, it would cause slightly more code to run on downlevels but I guess that doesn't matter.

brentfulgham commented on Source/WebCore/platform/sql/SQLiteTransaction.cpp:
```diff
@@ -27,10 +27,7 @@
 
 #include "SQLiteDatabase.h"
-
-#if PLATFORM(IOS_FAMILY)
```
Should this be `#if PLATFORM(COCOA)`? Or does this stuff exist/work on GTK/Windows?

brentfulgham replied:
Looking at the tracker code, it seems cross-platform, so this seems right.

cdumez replied:
The SQLiteDatabaseTracker builds fine on all platforms but is a no-op on some platforms. It doesn't hurt to run this code everywhere and dropping the `#if` checks makes the code nicer.

cdumez reviewed (approved):
LGTM

bnham committed:
Make SQLiteTransaction notify SQLiteDatabaseTracker on macOS
https://bugs.webkit.org/show_bug.cgi?id=259062
rdar://112027513

Reviewed by Chris Dumez and Brent Fulgham.

In 265791@main we enabled SQLiteDatabaseTracker on macOS, but it still does nothing because
SQLiteTransaction doesn't notify it. Change SQLiteTransaction to notify the tracker about ongoing
transactions on macOS as well.

* Source/WebCore/platform/sql/SQLiteTransaction.cpp:
(WebCore::SQLiteTransaction::begin):
(WebCore::SQLiteTransaction::commit):
(WebCore::SQLiteTransaction::rollback):
(WebCore::SQLiteTransaction::stop):

Canonical link: https://commits.webkit.org/265918@main
